pkg/controller/builder/image_builder.go

```diff
@@ -0,0 +1,114 @@
+package builder
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-activegate-operator/pkg/apis/dynatrace/v1alpha1"
+	_const "github.com/Dynatrace/dynatrace-activegate-operator/pkg/controller/const"
+	"github.com/Dynatrace/dynatrace-activegate-operator/pkg/dtclient"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateOrUpdateSecretIfNotExists creates a secret in case it does not exist or updates it if there are changes
+func CreateOrUpdateSecretIfNotExists(c client.Client, r client.Reader, secretName string, targetNS string, data map[string][]byte, secretType corev1.SecretType, log logr.Logger) error {
```

Comment:
(First, I'm not fond of a builder package, but that aside...)

`CreateOrUpdateSecretIfNotExists()` isn't a builder (i.e., constructor, or sort of.)

Reply:
I agree, it's more of a factory.

Reply:
Moved